src/main/python/verification/controllers.py

```diff
@@ -24,24 +37,69 @@ def verify_id():
         URL: http://localhost:5000/verifyID
         ----------------------------------------------------------------------
         """
-    # image_of_id = request.files.get("id_img")
-    # names = request.form['names']
-    # surname = request.form['surname']
-    # id_number = request.form['id_number']
-    # nationality = request.form['nationality']
-    # country_of_birth = request.form['cob']
-    # status = request.form['status']
-    # gender = request.form['gender']
-    # date_of_birth = request.form['dob']
-    # face = request.files.get('face_img')
-    logger.debug(request)
+    data = {"success": False}
+    # Get id image as numpy array
+    # Check to see if an image was uploaded.
+    if request.files.get("id_img", None) is not None:
+        # Grab the uploaded image.
+        image_of_id = _grab_image(stream=request.files["id_img"])
+    # Otherwise, assume that a URL was passed in.
+    else:
+        # Grab the URL from the request.
+        url = request.args.get("url", None)
+        # If the URL is None, then return an error.
+        if url is None:
+            data["error"] = "No URL provided."
+            return jsonify(data)
+            # Load the image and convert.
+        image_of_id = _grab_image(url=url)
 
-    # do stuff to get result
+    # Get face image as numpy array
+    # Check to see if an image was uploaded.
+    if request.files.get("face_img", None) is not None:
+        # Grab the uploaded image.
+        face = _grab_image(stream=request.files["face_img"])
+    # Otherwise, assume that a URL was passed in.
+    else:
+        # Grab the URL from the request.
+        url = request.args.get("url", None)
+        # If the URL is None, then return an error.
+        if url is None:
+            data["error"] = "No URL provided."
+            return jsonify(data)
+            # Load the image and convert.
+        face = _grab_image(url=url)
+
+    entered_details = {
+        "names": request.form['names'], "surname": request.form['surname'], "identity_number":
+        request.form['idNumber'], "nationality": request.form['nationality'], "country_of_birth":
+        request.form['cob'], "status": request.form['status'], "sex": request.form['gender'], "date_of_birth":
+        request.form['dob']
+    }
+
+    # Extract face
+    face_detector = FaceDetector(SHAPE_PREDICTOR_PATH)
+    gray_extracted_face1 = face_detector.face_likeness_extraction(image_of_id)
```

Comment:
(... follow-up comment on the one about the grayscale) Add the line here for converting the image to grayscale. The reason is because we don't want our code to do something that the user does not expect (based on the name of the function). It would therefore be something like:
```python
tempface = face_detector.face_likeness_extraction(image_of_id)
gray_extracted_face1 = cv2.cvtColor(tempface, cv2.COLOR_BGR2GRAY)
tempface = face_detector.face_likeness_extraction(face)
gray_extracted_face2 = cv2.cvtColor(tempface, cv2.COLOR_BGR2GRAY)
```